src/go/MONIT/monit.go

```diff
@@ -410,7 +456,7 @@ func queryURL(rurl string, headers [][]string, verbose int) Record {
 }
 
 // helper function to run ES/InfluxDB or MONIT query
-func run(rurl, token string, dbid int, dbname, query, esapi string, idx, limit, verbose int) Record {
+func run(rurl, token string, dbid int, dbname, query, esapi string, verbose int) Record {
```

Comment:
how about further improvement. You may pass StatsParams along with query to run function too instead of individual parameters, and this will simply code and make code further.

Reply:
I am not sure if this works, since `run` method is used not only with `StatsParams`, but also in a more general case for other queries (see [line 1076](https://github.com/dmwm/CMSMonitoring/blob/30bd92762e1c7c23faa453b11fbd64dd944e818e/src/go/MONIT/monit.go#L1076)).

Reply:
And, why you can't change this line:
```
data := run(url, t, dbid, database, q, esapi, verbose)
```
to something like this:
```
args := StatsParams{Token: t, ...}
data := run(args, q)
```
Bottom line that you don't need **all** parameters to initialize in StatsParams since it will be done automatically according to their data-type. You'll only need to initialize those which you'll use in run function with new signature which will use StatsParams and query. In fact, you may expand StatsParams to include other args, e.g. esapi which is not used by getStats but will be used by run function. You can also do this way:
```
// todo: adjust StatsParams to include esapi
args := NewStats(...)
args.ESAPI = esapi
data = run(args, q)
```
or if you want you can do this way:
```
args = StatsParams{Token: t, ...}
// if you define run function to have StatsParams, esapi and query then you can do
data = run(args, esapi, q)
```

Reply:
Ok, thank you for your advice!